core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java

```diff
@@ -238,14 +239,21 @@ private synchronized void initServer() {
       mStateMachine.close();
     }
     mStateMachine = new JournalStateMachine(mJournals, () -> this.getJournalSinks(null));
+    // Read external proxy configuration.
+    GrpcMessagingProxy serverProxy = null;
+    if (mConf.getProxyAddress() != null) {
+      serverProxy = new GrpcMessagingProxy();
+      serverProxy.addProxy(getLocalAddress(mConf), new Address(mConf.getProxyAddress()));
+    }
```

Comment:
Consider creating the `GrpcMessagingTransport` here? Then we won't have to do as many null checks here and in `GrpcMessagingTransport` since we can assume the proxy is only set if necessary.

Reply:
I was able to strip some null checks by enforcing valid proxy reference.